### PR TITLE
Role based authorization implemented

### PR DESCRIPTION
**Pull request by p1q**

### CODE REVIEW

File: src/main/java/internetshop/controllers/LogoutController.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class LogoutController extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest request,
+                         HttpServletResponse response) throws IOException {
+        HttpSession session = request.getSession(true);
```
Comment (Sofasmile) [inline src/main/java/internetshop/controllers/LogoutController.java]: HttpSession session = request.getSession(true);
        session.setAttribute("userId", "");
-->
request.getSession().invalidate()

File: src/main/java/internetshop/filters/AuthorizationFilter.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
```
Comment (Sofasmile) [inline src/main/java/internetshop/filters/AuthorizationFilter.java]: remove two redundant lines

File: src/main/java/internetshop/filters/AuthorizationFilter.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    private static UserService userService;
+
+    private Map<String, RoleName> protectedAdminUrls = new HashMap<>();
+    private Map<String, RoleName> protectedUserUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedAdminUrls.put("/user/add-item", ADMIN);
+        protectedAdminUrls.put("/user/added-item", ADMIN);
+        protectedAdminUrls.put("/user/add-user", ADMIN);
```
Comment (Sofasmile) [inline src/main/java/internetshop/filters/AuthorizationFilter.java]: what is difference between `/user/added-item` and `/user/add-user`?

Comment (p1q) [reply src/main/java/internetshop/filters/AuthorizationFilter.java]: /user/added-item is a page to show result of /user/add-item

File: src/main/java/internetshop/filters/AuthorizationFilter.java (hunk 2 above)
Comment (p1q) [reply src/main/java/internetshop/filters/AuthorizationFilter.java]: Done

**Pull request merged**